nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-component-version.js

```diff
@@ -0,0 +1,340 @@
+        }
+    };
+
+    /**
+     * Gets the field to use to access the returned types/bundles.
+     *
+     * @param {object} componentEntity
+     * @returns {string} field
+     */
+    var getTypeField = function (componentEntity) {
+        if (componentEntity.type === 'ReportingTask') {
+            return 'reportingTaskTypes';
+        } else if (componentEntity.type === 'ControllerService') {
+            return 'controllerServiceTypes';
+        } else {
+            return 'processorTypes';
+        }
+    };
+
+    /**
+     * Reset the dialog.
+     */
+    var resetDialog = function () {
+        // clear the versions
+        var versions = versionMap.keys();
+        $.each(versions, function (_, version) {
+            versionMap.remove(version);
+        });
+
+        // clear the service apis
+        $('#component-version-controller-service-apis').empty();
+        $('#component-version-controller-service-apis-container').hide();
+
+        // clear the fields
+        $('#component-version-name').text('');
+        $('#component-version-bundle').text('');
+        $('#component-version-tags').text('');
+        $('#component-version-restriction').removeClass('unset').text('');
+        $('#component-version-description').text('');
+
+        // destroy the version combo
+        $('#component-version-selector').combo('destroy');
+
+        // removed the stored data
+        $('#component-version-dialog').removeData('component');
+    };
+
+    /**
+     * Sets the specified option.
+     *
+     * @param {object} selectedOption
+     */
+    var select = function (selectedOption) {
+        var documentedType = versionMap.get(selectedOption.value);
+
+        // set any restriction
+        if (nfCommon.isDefinedAndNotNull(documentedType.usageRestriction)) {
+            $('#component-version-restriction').text(documentedType.usageRestriction);
+        } else {
+            $('#component-version-restriction').addClass('unset').text('No restriction');
+        }
+
+        // update the service apis if necessary
+        if (!nfCommon.isEmpty(documentedType.controllerServiceApis)) {
+            var formattedControllerServiceApis = nfCommon.getFormattedServiceApis(documentedType.controllerServiceApis);
+            var serviceTips = nfCommon.formatUnorderedList(formattedControllerServiceApis);
+            $('#component-version-controller-service-apis').empty().append(serviceTips);
+            $('#component-version-controller-service-apis-container').show();
+        }
+
+        // update the tags and description
+        $('#component-version-tags').text(documentedType.tags.join(', '));
+        $('#component-version-description').text(documentedType.description);
+    };
+
+    return {
+        init: function (settings, processGroupConfiguration) {
+            versionMap = d3.map();
+            nfSettings = settings;
+            nfProcessGroupConfiguration = processGroupConfiguration;
```

Comment:
Should be able to remove the injection of the nfProcessGroupConfiguration module here too...